### PR TITLE
feat(react-drawer): dialog slot to be only used by internal composition API

### PR DESCRIPTION
**Pull request by marcosmoura**
## Previous Behavior
Dialog was created as an external slot and users could easily override to any other component with the same signature. That could lead to a completely broken Drawer as it is heavily dependent on Dialog.

## New Behavior
Makes the Dialog as an internal slot. That makes it possible to override the Dialog component only when explicitly using the composition API.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Avatar-mount_base.html">615</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Avatar-mount.html">655</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Button-mount_base.html">302</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Button-mount.html">294</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Field-mount_base.html">1102</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Field-mount.html">1124</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProvider-mount_base.html">696</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProvider-mount.html">691</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProviderWithTheme-mount_base.html">79</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProviderWithTheme-mount.html">85</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">68</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/MakeStyles-mount_base.html">860</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/MakeStyles-mount.html">865</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Persona-mount_base.html">1723</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/Persona-mount.html">1709</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/SpinButton-mount_base.html">1405</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29279/perf-test-react-components/SpinButton-mount.html">1378</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.572 kB`<br />`19.658 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.688 kB`<br />`59.552 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `40.966 kB`<br />`13.569 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.541 kB`<br />`2.227 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c1ae0fb42575ae2deb11b32110538adbd0f652ad'>c1ae0fb42575ae2deb11b32110538adbd0f652ad</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-drawer-840c300d-046b-4ec7-b733-f88067df7dea.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "feat: make dialog slot internal to be used for composition only",
+  "packageName": "@fluentui/react-drawer",
+  "email": "marcosvmmoura@gmail.com",
+  "dependentChangeType": "patch"
```
Comment (ValentinaKozlova) [inline change/@fluentui-react-drawer-840c300d-046b-4ec7-b733-f88067df7dea.json:6]: Don't forget to update in to feature

**Review by ValentinaKozlova — APPROVED** [review]

**Comment by marcosmoura** [comment]
This PR is not needed as https://github.com/microsoft/fluentui/pull/29392 now have all of these changes.

**Pull request closed without merging**